helpers_test.go

```diff
@@ -29,4 +29,6 @@ func TestDecodeResponse(t *testing.T) {
 	assert.Equal(t, expected, r)
 
 	assert.Error(t, DecodeResponse([]string{"line1", "line2"}, r))
+	assert.Error(t, DecodeResponse([]string{}, r))
```

Comment:
its not immediately clear to why this is needed, so likely needs a comment to explain

Reply:
Well, calling the decoder with an empty array previously also resulted in an error (confusingly it returned: "too many lines"). That's why I added the new error and the new test case to assert that no input returns an error.
So... should I write a comment that says "Calling the decoder with no input should result in an error"?

Reply:
Ahh right your testing the other error condition, didn't spot that.

Might be better to change it to an EqualError test so its clear what each bit is testing.